File: internal/model/entity/menu.go
package entity

import "github.com/gogf/gf/v2/os/gtime"

type Menu struct {
    Id        uint64      `orm:"id"`
    ParentId  uint64      `orm:"parent_id"`
    Name      string      `orm:"name"`
    Path      string      `orm:"path"`
    Component string      `orm:"component"`
    Icon      string      `orm:"icon"`
    Sort      int         `orm:"sort"`
    Visible   uint        `orm:"visible"`
    Status    uint        `orm:"status"`
    Type      uint        `orm:"type"`
    CreatedAt *gtime.Time `orm:"created_at"`
    UpdatedAt *gtime.Time `orm:"updated_at"`
    DeletedAt *gtime.Time `orm:"deleted_at"`
}

File: web/src/api/menu.ts
import request from '@/utils/request'

export interface MenuTree {
  id: number
  parent_id: number
  name: string
  path: string
  component: string
  icon: string
  sort: number
  visible: number
  type: number
  children?: MenuTree[]
}

export interface MenuCreateReq {
  parent_id?: number
  name: string
  path?: string
  component?: string
  icon?: string
  sort?: number
  visible?: number
  status?: number
  type?: number
}

export interface MenuItem {
  id: number
  parent_id: number
  name: string
  path: string
  component: string
  icon: string
  sort: number
  visible: number
  status: number
  type: number
  created_at: string
}

export interface MenuListRes {
  list: MenuItem[]
  total: number
  page: number
}

export const menuApi = {
  create: (data: MenuCreateReq) => request.post('/menu', data),
  delete: (id: number) => request.delete(`/menu/${id}`),
  update: (id: number, data: Partial<MenuCreateReq>) => request.put(`/menu/${id}`, data),
  getOne: (id: number) => request.get(`/menu/${id}`),
  list: (params: { page: number; page_size: number; name?: string; status?: number }) =>
    request.get('/menu', { params }),
  getUserMenus: () => request.get<{ menus: MenuTree[] }>('/menu/user'),
}
